### ISSUE TITLE
support socks5 client+server

### ISSUE THREAD

**Issue by GlenDC** [plabayo/rama#15]

**Comment by GlenDC** [comment]
Might be a good idea to fork the code from https://github.com/EAimTY/socks5-server directly into here, with appropriate attributions in a text file of that module. Because the repo is a bit inactive in terms of responding to issues/PRs, and does not directly fit into the needs of rama.

That combined with the RFCs, should be sufficient to add support here.

**Comment by pm100** [comment]
note that https://github.com/EAimTY/socks5-server is GPL, not compatible with your license

**Comment by GlenDC** [comment]
Correct, thank you for pointing out. This is however an old issue and since than I already have studied the corresponding RFC's extensively. So once I (or someone else) has time for it we can probably start from scratch given it is only really the protocol code which isn't that complex to begin with. Or start from another library, either way the code has to be embedded / written as `rama-socks5` so we have full control over it.

**Comment by GlenDC** [comment]
RFC's:

- <https://datatracker.ietf.org/doc/html/rfc1928>
- <https://datatracker.ietf.org/doc/html/rfc1929>

Relevant IANA registries:

- <https://www.iana.org/assignments/socks-methods/socks-methods.xhtml>
